vsb/databases/opensearch/opensearch.py

```diff
@@ -24,15 +24,24 @@ def __init__(
         self.namespace = namespace
 
     def insert_batch(self, batch: RecordList):
-        actions = []
-        action = {"index": {"_index": self.index_name}}
-        for rec in batch:
-            vector_document = {"vsb_vec_id": rec.id, "v_content": np.array(rec.values)}
-            actions.append(action)
-            actions.append(vector_document)
-
-        # Bulk ingest documents
-        upload_response = self.client.bulk(body=actions)
+        @retry(
+            wait=wait_exponential_jitter(initial=0.1, jitter=0.1),
+            stop=stop_after_attempt(5),
+            after=after_log(logger, logging.DEBUG),
+        )
+        def do_insert_with_retry():
+            actions = []
+            action = {"index": {"_index": self.index_name}}
+            for rec in batch:
+                vector_document = {"vsb_vec_id": rec.id, "v_content": np.array(rec.values)}
+                actions.append(action)
+                actions.append(vector_document)
+            # Bulk ingest documents
+            return self.client.bulk(body=actions,request_timeout=600)
```

Comment:
Note: You don't need to retry the creation of the actions list - just move that into the head of `insert_batch`, and then have your `do_insert_with_retry` method just call `self.client.bulk`.